enhancements/machine-config/securing-mcs.md

```diff
@@ -0,0 +1,112 @@
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The bootstrapping process for new cluster nodes is a tricky problem, primarily due to the fact that a new nodes start from a position of almost zero knowledge. These nodes need to fetch their machine configuration and authenticate themselves with the cluster before they can access most resources and have workloads scheduled. At the same time, a malicious actor pretending to be a new node cannot be allowed to join the cluster. Otherwise, they would be able to get access to sensitive resources and could have workloads and their secrets scheduled to them. Today, we protect against this by preventing cluster workloads from accessing the Machine Config Server, the component of the Machine Config Operator which is responsible for serving Ignition configs, derived from a set of Machine Configs, to new machines. This approach presents a problem though: workloads on OpenShift are not allowed to make use of TCP ports 22623 and 22624, the ports used by the Machine Config Server.
+
+## Motivation
+
+This blanket network policy prevents traffic destined for ports 22623 and 22624, regardless of the source and destination IP addresses. This has the potential to impact customer workloads, but also internal initiatives (e.g. OpenShift on OpenShift is complicated by this restriction).
+
+### Goals
+
+- Remove the port restriction on 22623 and 22624
+- Ensure that the node bootstrap flow remains secure
+
+### Non-Goals
+
+- Allow customers to use Machine Configs to provide pre-bootstrap configuration
+
+## Proposal
+
+If it can be ensured that no one is able to access sensitive data through the Machine Config Server, the port restrictions can be safely removed. In order to achieve this, Machine Configs can be split into two groups: pre-node-approved configs and post-node-approved configs. The pre-node-approved configs are ones which do not contain sensitive information and contain just enough to configure new nodes to submit a certificate signing request (CSR) to the API server. The post-node-approved configs contain everything else, including the overwhelming majority of customer-provided Machine Configs. When a new machine is created, it fetches its Ignition config from the Machine Config Server, which only serves the pre-node-approved configs. At this point, the machine has enough information to submit a CSR, but nothing sensitive (note that for the purposes of this proposal, the CSR bootstrap token and pull secret are not considered sensitive). Once the CSR has been approved and a kubeconfig returned, the Machine Config Daemon is able to directly enumerate Machine Configs, including the post-node-approved ones, so it then applies any remaining configuration to the machine and reboots if necessary. At this point, the machine is fully configured.
```

Comment:
I think it would be helpful to pull out some of the assumptions and state them very clearly up front.  It's included in here, but perhaps before this paragraph, a new paragraph could say:

Security assumptions:
* The CSR bootstrap token is not considered sensitive and being accessible to all applications in the cluster is accessible.
* The pull secret is not considered sensitive and being accessible to all applications in the cluster is accessible
* Contents of MachineConfig resources are considered (potentially) sensitive, so this proposal introduces a change to protect them from unauthorized access through the MCS

Do I have that right?

This proposal would have to depend on some other enhancement that would allow that first assumption to hold.  I believe right now you could DoS a cluster with the ability to issue CSRs.